src/huak/config/pyproject/project/mod.rs

```diff
@@ -27,7 +28,7 @@ pub(crate) struct Project {
     pub(crate) description: String,
     pub(crate) dependencies: Vec<String>,
     #[serde(rename = "optional-dependencies")]
-    pub(crate) optional_dependencies: Option<Vec<String>>,
+    pub(crate) optional_dependencies: Option<HashMap<String, Vec<String>>>,
```

Comment:
I'm not 100% on this, but since we have #247 coming I'm less concerned. But I think there's `Map` from `toml`. If this works and it makes sense to use we can keep it. Just wanted to make the callout.

Reply:
Apparently they are also using [HashMap](https://github.com/PyO3/pyproject-toml-rs/blob/66b1592675f19810c7db79629071e6231006b134/src/lib.rs#L61) so I think we are good.